utils/python/CIME/XML/generic_xml.py

```diff
@@ -51,11 +51,12 @@ def write(self, infile=None):
         """
         Write an xml file from data in self
         """
+        if(infile is None):
+            infile = self.filename
         logging.info("write: "+ infile)
-        if(infile != None):
```

Comment:
Yes, when I tried to output the internally generated testspec.xml file it was alll on a single line with no formatting.  I would really like to use the xslt library here but the python interface doesn't seem to be generally available.